packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```
@@ -0,0 +1,1368 @@
+          escrowAddress,
+          recipients,
+          amounts,
+          finalResultsUrl,
+          finalResultsHash
+        )
+      ).rejects.toThrow(ErrorEscrowDoesNotHaveEnoughBalance);
+    });
+
+    test('should successfully bulkPayOut escrow', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const recipients = [
+        ethers.constants.AddressZero,
+        ethers.constants.AddressZero,
+      ];
+      const amounts = [BigNumber.from(10), BigNumber.from(10)];
+      const finalResultsUrl = VALID_URL;
+      const finalResultsHash = FAKE_HASH;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.getBalance = vi.fn().mockReturnValue(BigNumber.from(100));
+
+      await escrowClient.bulkPayOut(
+        escrowAddress,
+        recipients,
+        amounts,
+        finalResultsUrl,
+        finalResultsHash
+      );
+
+      expect(escrowClient.escrowContract.bulkPayOut).toHaveBeenCalledWith(
+        recipients,
+        amounts,
+        finalResultsUrl,
+        finalResultsHash,
+        DEFAULT_TX_ID
+      );
+    });
+
+    test('should throw an error if the bulkPayOut fails', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.escrowContract.abort.mockRejectedValueOnce(new Error());
+
+      await expect(escrowClient.abort(escrowAddress)).rejects.toThrow();
+
+      expect(escrowClient.escrowContract.abort).toHaveBeenCalledWith();
+    });
+  });
+
+  describe('cancel', () => {
+    test('should throw an error if escrowAddress is an invalid address', async () => {
+      const invalidAddress = FAKE_ADDRESS;
+
+      await expect(escrowClient.cancel(invalidAddress)).rejects.toThrow(
+        ErrorInvalidEscrowAddressProvided
+      );
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
```

fix writing:
'should throw an error if hasEscrow returns false'